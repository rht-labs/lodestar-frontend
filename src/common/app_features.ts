// All roles defined by the backend to be used in the frontend should be enumerated in this object.

export const APP_FEATURES = {
  admin: 'admin',
  writer: 'writer',
  reader: 'reader',
  notifications: 'notifications',
  engagementCardIcons: 'engagementCardIcons',
  resetUser: 'resetUser',
};

export type AppFeature = keyof typeof APP_FEATURES;
