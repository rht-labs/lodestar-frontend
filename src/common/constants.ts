export const APPLICATION_CONTENT_CONTAINER_ID = 'main-application-content';
