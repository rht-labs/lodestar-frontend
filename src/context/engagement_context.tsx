import React, { createContext, useEffect, useState, useCallback } from 'react';
import { Engagement } from '../models/engagement';
import { EngagementRepository } from '../repositories/engagement/engagement_repository';

export interface EngagementContext {
  getEngagements: () => void;
  engagements: Engagement[];
}

export const EngagementContext = createContext<EngagementContext>({
  getEngagements: async () => [],
  engagements: [],
});
const { Provider } = EngagementContext;

export const EngagementProvider = ({
  children,
  engagementRepository,
}: {
  children: React.ReactChild;
  engagementRepository: EngagementRepository;
}) => {
  const [engagements, setEngagements] = useState<Engagement[]>([]);

  const fetchEngagements = useCallback(async () => {
    const engagements = await engagementRepository.fetchEngagements();
    setEngagements(engagements);
  }, [engagementRepository]);

  useEffect(() => {
    fetchEngagements();
  }, [fetchEngagements]);

  return (
    <Provider
      value={{
        engagements,
        getEngagements: fetchEngagements,
      }}
    >
      {children}
    </Provider>
  );
};
