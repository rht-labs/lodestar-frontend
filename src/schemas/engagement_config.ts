import FakedSchema from './fixtures/engagement_form_config.json';

export interface EngagementFormOption {
  label: string;
  value: string;
  options?: EngagementFormOption[];
}

export interface EngagementFormValidator {
  kind: string;
  value?: any;
}

export interface FormConfig {
  options?: EngagementFormOption[];
  validators?: EngagementFormValidator[];
}

export interface EngagementFormConfig {
  [key: string]: { [key: string]: FormConfig };
}

export abstract class EngagementFormConfig {
  static fromFake(): EngagementFormConfig {
    return FakedSchema;
  }
}
