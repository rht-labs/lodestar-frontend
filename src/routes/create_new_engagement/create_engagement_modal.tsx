import React, { useEffect, useState } from 'react';
import {
  Modal,
  Button,
  Form,
  FormGroup,
  TextInput,
  Text,
} from '@patternfly/react-core';
import { useEngagements } from '../../context/engagement_context/engagement_hook';
import { CustomerSelectDropdown } from '../../components/customer_select_dropdown/customer_select_dropdown';
import { useValidation } from '../../context/validation_context/validation_hook';

export interface CreateEngagementModalProps {
  isOpen: boolean;
  onRequestClose: (customerName?: string, projectName?: string) => void;
}

export function CreateEngagementModal(props: CreateEngagementModalProps) {
  const { currentEngagement, createEngagement } = useEngagements();

  const [customerName, setCustomerName] = useState(null);
  const [projectName, setProjectName] = useState(null);
  const { getValidationResult, validate } = useValidation();
  useEffect(() => {
    setCustomerName(null);
    setProjectName(null);
  }, [currentEngagement]);

  const createNewEngagement = () => {
    createEngagement({
      customer_name: customerName,
      project_name: projectName,
    });
    props.onRequestClose(customerName, projectName);
  };
  const hasValidInput =
    !!customerName &&
    !!projectName &&
    getValidationResult('project_name')?.length === 0 &&
    getValidationResult('customer_name')?.length === 0;
  return (
    <Modal
      width={'50%'}
      title="Create New Engagement"
      isOpen={props.isOpen}
      onClose={props.onRequestClose}
      actions={[
        <Button
          key="confirm"
          variant="primary"
          onClick={createNewEngagement}
          data-cy="createNewEngagement"
          isDisabled={!hasValidInput}
        >
          Submit
        </Button>,
        <Button
          key="cancel"
          variant="link"
          onClick={() => props.onRequestClose()}
          data-cy={'cancel_button'}
        >
          Cancel
        </Button>,
      ]}
    >
      <Text component="h6">
        To create a new Engagement, please enter a client and product name then
        click submit.
      </Text>
      <Form isHorizontal>
        <FormGroup
          label="Customer Name"
          fieldId="customer-name"
          helperText="What client is this for?"
          isRequired
          helperTextInvalid={getValidationResult('customer_name').join(' ')}
          validated={
            getValidationResult('customer_name').length > 0
              ? 'error'
              : 'default'
          }
        >
          <CustomerSelectDropdown
            placeholder="e.g. NASA"
            selectedValue={customerName}
            onSelect={value =>
              validate('customer_name')(value) && setCustomerName(value)
            }
          />
        </FormGroup>
        <FormGroup
          label="Project Name"
          fieldId="project-name"
          helperText="The name of the solution being worked on."
          isRequired
          helperTextInvalid={getValidationResult('project_name').join(' ')}
          validated={
            getValidationResult('project_name').length > 0 ? 'error' : 'default'
          }
        >
          <TextInput
            type="text"
            id="project_name"
            name="project_name"
            placeholder="e.g. Mars Probe"
            value={projectName || ''}
            onChange={value =>
              validate('project_name')(value) && setProjectName(value)
            }
            data-cy="new_engagement_name"
          />
        </FormGroup>
      </Form>
    </Modal>
  );
}
