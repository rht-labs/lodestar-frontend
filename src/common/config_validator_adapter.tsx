import { ValidatorFactory } from '../schemas/validators';
import { EngagementFormConfig } from '../schemas/engagement_config';

export const getValidatorsFromFormOptions = (
  formOptions: EngagementFormConfig = {}
) =>
  Object.keys(formOptions || {}).reduce(
    (previousSchemaGroups, currentGroupKey) => {
      return {
        ...previousSchemaGroups,
        ...Object.keys(formOptions[currentGroupKey] ?? {}).reduce(
          (previousGroupFields, currentFieldKey) => ({
            ...previousGroupFields,
            [currentFieldKey]: (
              formOptions?.[currentGroupKey]?.[currentFieldKey]?.validators ||
              []
            ).map(ValidatorFactory),
          }),
          {}
        ),
      };
    },
    {}
  );
