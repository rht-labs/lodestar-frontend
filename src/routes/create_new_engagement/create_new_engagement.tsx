import React from 'react';
import { CreateEngagementModal } from './create_engagement_modal';
import { useHistory } from 'react-router';
import { EngagementDetailViewContainer } from '../engagement_details/engagement_details';
import { ValidationProvider } from '../../context/validation_context/validation_context';
import { getValidatorsFromFormOptions } from '../../common/config_validator_adapter';
import { useEngagements } from '../../context/engagement_context/engagement_hook';

export function CreateNewEngagement(props) {
  const history = useHistory();
  const handleModalClose = (customerName, projectName) => {
    if (customerName && projectName) {
      history.push(`/app/engagements/${customerName}/${projectName}`);
    } else {
      history.push('/app/engagements');
    }
  };

  const { formOptions } = useEngagements();

  return (
    <div>
      <ValidationProvider
        validators={getValidatorsFromFormOptions(formOptions)}
      >
        <CreateEngagementModal
          onRequestClose={handleModalClose}
          isOpen={true}
        />
      </ValidationProvider>
      <EngagementDetailViewContainer />
    </div>
  );
}
