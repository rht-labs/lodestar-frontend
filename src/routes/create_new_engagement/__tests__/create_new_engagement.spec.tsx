import React from 'react';
import { CreateNewEngagement } from '../create_new_engagement';
import { TestStateWrapper } from '../../../common/test_state_wrapper';
import {
  render,
  act,
  fireEvent,
  queryByAttribute,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { EngagementContext } from '../../../context/engagement_context/engagement_context';
import { EngagementFormConfig } from '../../../schemas/engagement_config';
const getById = queryByAttribute.bind(null, 'id');

describe('Create New Engagement Route', () => {
  const getComponent = () => {
    return (
      <MemoryRouter>
        <TestStateWrapper>
          <CreateNewEngagement />
        </TestStateWrapper>
      </MemoryRouter>
    );
  };
  test('should match snapshot', async () => {
    await act(async () => {
      const rendered = render(getComponent());
      expect(rendered).toMatchSnapshot();
    });
  });
  test('should have a field for a project name', () => {
    const wrapper = render(getComponent());
    expect(wrapper.getByTestId('project-name')).toBeDefined();
  });
  test('should have a field for a customer name', () => {
    const wrapper = render(getComponent());
    expect(wrapper.getByTestId('customer-name')).toBeDefined();
  });
  test('should have a field for a region name', () => {
    const wrapper = render(getComponent());
    expect(wrapper.getByTestId('region')).toBeDefined();
  });
  test('clicking the submit button creates a new engagement', async () => {
    const createEngagement = jest.fn();
    const Component = () => (
      <MemoryRouter>
        <EngagementContext.Provider
          value={{
            engagements: [{ customer_name: 'a' }],
            getEngagements: async () => [],
            createEngagement,
            getConfig: () => {},
            formOptions: EngagementFormConfig.fromFake(),
          }}
        >
          <CreateNewEngagement />
        </EngagementContext.Provider>
      </MemoryRouter>
    );

    const wrapper = render(<Component />);
    await act(async () => {
      const customerNameField = wrapper
        .getByTestId('customer-name')
        .getElementsByTagName('input')[0];
      await fireEvent.change(customerNameField, {
        target: { value: 'a' },
      });
      const dropdownButton = getById(wrapper.container, 'customer_dropdown');
      fireEvent.click(dropdownButton);
      wrapper.rerender(<Component />);
      await fireEvent.click(wrapper.getByTestId('a'));
      await fireEvent.change(wrapper.getByTestId('project-name'), {
        target: { value: 'Mars Rover' },
      });
      await fireEvent.change(wrapper.getByTestId('region'), {
        target: { value: 'na' },
      });
      wrapper.rerender(<Component />);
      const createButton = wrapper.getByTestId('create-engagement-button');
      fireEvent.click(createButton);
    });
    expect(createEngagement).toHaveBeenCalled();
  });
});
