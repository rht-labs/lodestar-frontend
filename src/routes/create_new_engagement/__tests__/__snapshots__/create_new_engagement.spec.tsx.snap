// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create New Engagement Route should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="min-height: 100%; display: flex; flex-direction: column;"
      >
        <section
          class="pf-c-page__main-section pf-m-light"
        >
          <h1
            class="pf-c-title pf-m-2xl"
          >
            Create New Engagement
          </h1>
          <h6
            class=""
            data-pf-content="true"
          >
            To create a new Engagement, please enter a client and product name then click submit.
          </h6>
        </section>
        <section
          class="pf-c-page__main-section pf-m-fill"
        >
          <div
            class="pf-l-grid"
          >
            <div
              class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
            >
              <form
                class="pf-c-form"
                novalidate=""
              >
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="customer-name"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        <div
                          style="display: inline-block;"
                        >
                          <h2
                            class="pf-c-title pf-m-xl"
                          >
                            Client Name  
                            <svg
                              aria-hidden="true"
                              fill="lightgrey"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 512 512"
                              width="1em"
                            >
                              <path
                                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                                transform=""
                              />
                            </svg>
                          </h2>
                        </div>
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <div
                      data-testid="customer-name"
                    >
                      <div
                        class="pf-c-select"
                        data-ouia-component-id="0"
                        data-ouia-component-type="PF4/Select"
                        data-ouia-safe="true"
                      >
                        <div
                          class="pf-c-select__toggle pf-m-typeahead"
                        >
                          <div
                            class="pf-c-select__toggle-wrapper"
                          >
                            <input
                              aria-label=""
                              autocomplete="off"
                              class="pf-c-form-control pf-c-select__toggle-typeahead"
                              id="customer_dropdown-select-typeahead"
                              placeholder="NASA"
                              type="text"
                              value=""
                            />
                          </div>
                          <button
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-label="Options menu"
                            aria-labelledby=" customer_dropdown"
                            class="pf-c-button pf-c-select__toggle-button pf-m-plain"
                            id="customer_dropdown"
                            tabindex="-1"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="pf-c-select__toggle-arrow"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 320 512"
                              width="1em"
                            >
                              <path
                                d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                                transform=""
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="project-name"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        <div
                          style="display: inline-block;"
                        >
                          <h2
                            class="pf-c-title pf-m-xl"
                          >
                            Engagement Name  
                            <svg
                              aria-hidden="true"
                              fill="lightgrey"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 512 512"
                              width="1em"
                            >
                              <path
                                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                                transform=""
                              />
                            </svg>
                          </h2>
                        </div>
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      class="pf-c-form-control"
                      data-cy="new_engagement_name"
                      data-testid="project-name"
                      id="project_name"
                      name="project_name"
                      placeholder="Mars Probe"
                      type="text"
                      value=""
                    />
                    <div
                      aria-live="polite"
                      class="pf-c-form__helper-text"
                      id="project-name-helper"
                    >
                      A descriptive name to idenitfy your engagement
                    </div>
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="region"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        <div
                          style="display: inline-block;"
                        >
                          <h2
                            class="pf-c-title pf-m-xl"
                          >
                            Region  
                            <svg
                              aria-hidden="true"
                              fill="lightgrey"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 512 512"
                              width="1em"
                            >
                              <path
                                d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                                transform=""
                              />
                            </svg>
                          </h2>
                        </div>
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <select
                      aria-invalid="false"
                      class="pf-c-form-control"
                      data-ouia-component-id="1"
                      data-ouia-component-type="PF4/FormSelect"
                      data-ouia-safe="true"
                      data-testid="region"
                    >
                      <option
                        class=""
                        value=""
                      >
                        Select a region
                      </option>
                    </select>
                    <div
                      aria-live="polite"
                      class="pf-c-form__helper-text"
                      id="region-helper"
                    >
                      The region sponsoring the engagement
                    </div>
                  </div>
                </div>
              </form>
              <button
                aria-disabled="true"
                class="pf-c-button pf-m-primary pf-m-disabled"
                data-ouia-component-id="2"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                data-testid="create-engagement-button"
                disabled=""
                style="margin: 1rem 0px;"
                type="button"
              >
                Submit
              </button>
            </div>
          </div>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="min-height: 100%; display: flex; flex-direction: column;"
    >
      <section
        class="pf-c-page__main-section pf-m-light"
      >
        <h1
          class="pf-c-title pf-m-2xl"
        >
          Create New Engagement
        </h1>
        <h6
          class=""
          data-pf-content="true"
        >
          To create a new Engagement, please enter a client and product name then click submit.
        </h6>
      </section>
      <section
        class="pf-c-page__main-section pf-m-fill"
      >
        <div
          class="pf-l-grid"
        >
          <div
            class="pf-l-grid__item pf-m-12-col-on-md pf-m-6-col-on-lg"
          >
            <form
              class="pf-c-form"
              novalidate=""
            >
              <div
                class="pf-c-form__group"
              >
                <div
                  class="pf-c-form__group-label"
                >
                  <label
                    class="pf-c-form__label"
                    for="customer-name"
                  >
                    <span
                      class="pf-c-form__label-text"
                    >
                      <div
                        style="display: inline-block;"
                      >
                        <h2
                          class="pf-c-title pf-m-xl"
                        >
                          Client Name  
                          <svg
                            aria-hidden="true"
                            fill="lightgrey"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 512 512"
                            width="1em"
                          >
                            <path
                              d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                              transform=""
                            />
                          </svg>
                        </h2>
                      </div>
                    </span>
                  </label>
                   
                </div>
                <div
                  class="pf-c-form__group-control"
                >
                  <div
                    data-testid="customer-name"
                  >
                    <div
                      class="pf-c-select"
                      data-ouia-component-id="0"
                      data-ouia-component-type="PF4/Select"
                      data-ouia-safe="true"
                    >
                      <div
                        class="pf-c-select__toggle pf-m-typeahead"
                      >
                        <div
                          class="pf-c-select__toggle-wrapper"
                        >
                          <input
                            aria-label=""
                            autocomplete="off"
                            class="pf-c-form-control pf-c-select__toggle-typeahead"
                            id="customer_dropdown-select-typeahead"
                            placeholder="NASA"
                            type="text"
                            value=""
                          />
                        </div>
                        <button
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-label="Options menu"
                          aria-labelledby=" customer_dropdown"
                          class="pf-c-button pf-c-select__toggle-button pf-m-plain"
                          id="customer_dropdown"
                          tabindex="-1"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="pf-c-select__toggle-arrow"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 320 512"
                            width="1em"
                          >
                            <path
                              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                              transform=""
                            />
                          </svg>
                        </button>
                      </div>
                    </div>
                  </div>
                  
                </div>
              </div>
              <div
                class="pf-c-form__group"
              >
                <div
                  class="pf-c-form__group-label"
                >
                  <label
                    class="pf-c-form__label"
                    for="project-name"
                  >
                    <span
                      class="pf-c-form__label-text"
                    >
                      <div
                        style="display: inline-block;"
                      >
                        <h2
                          class="pf-c-title pf-m-xl"
                        >
                          Engagement Name  
                          <svg
                            aria-hidden="true"
                            fill="lightgrey"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 512 512"
                            width="1em"
                          >
                            <path
                              d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                              transform=""
                            />
                          </svg>
                        </h2>
                      </div>
                    </span>
                  </label>
                   
                </div>
                <div
                  class="pf-c-form__group-control"
                >
                  <input
                    aria-invalid="false"
                    class="pf-c-form-control"
                    data-cy="new_engagement_name"
                    data-testid="project-name"
                    id="project_name"
                    name="project_name"
                    placeholder="Mars Probe"
                    type="text"
                    value=""
                  />
                  <div
                    aria-live="polite"
                    class="pf-c-form__helper-text"
                    id="project-name-helper"
                  >
                    A descriptive name to idenitfy your engagement
                  </div>
                </div>
              </div>
              <div
                class="pf-c-form__group"
              >
                <div
                  class="pf-c-form__group-label"
                >
                  <label
                    class="pf-c-form__label"
                    for="region"
                  >
                    <span
                      class="pf-c-form__label-text"
                    >
                      <div
                        style="display: inline-block;"
                      >
                        <h2
                          class="pf-c-title pf-m-xl"
                        >
                          Region  
                          <svg
                            aria-hidden="true"
                            fill="lightgrey"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 512 512"
                            width="1em"
                          >
                            <path
                              d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                              transform=""
                            />
                          </svg>
                        </h2>
                      </div>
                    </span>
                  </label>
                   
                </div>
                <div
                  class="pf-c-form__group-control"
                >
                  <select
                    aria-invalid="false"
                    class="pf-c-form-control"
                    data-ouia-component-id="1"
                    data-ouia-component-type="PF4/FormSelect"
                    data-ouia-safe="true"
                    data-testid="region"
                  >
                    <option
                      class=""
                      value=""
                    >
                      Select a region
                    </option>
                  </select>
                  <div
                    aria-live="polite"
                    class="pf-c-form__helper-text"
                    id="region-helper"
                  >
                    The region sponsoring the engagement
                  </div>
                </div>
              </div>
            </form>
            <button
              aria-disabled="true"
              class="pf-c-button pf-m-primary pf-m-disabled"
              data-ouia-component-id="2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              data-testid="create-engagement-button"
              disabled=""
              style="margin: 1rem 0px;"
              type="button"
            >
              Submit
            </button>
          </div>
        </div>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
